File: apps/portal/src/lib/api/schemas/inspection.ts
import { z } from 'zod';

export const InspectionVerdictEnum = z.enum(['pass', 'fail', 'not_applicable']);
export type InspectionVerdictValue = z.infer<typeof InspectionVerdictEnum>;

export const ChecklistItemResultSchema = z.object({
  id: z.string().uuid(),
  checklist_item_id: z.string().uuid(),
  borgingsmoment_id: z.string().uuid(),
  project_id: z.string().uuid(),
  verdict: InspectionVerdictEnum,
  note: z.union([z.string(), z.null()]),
  inspector_user_id: z.string().uuid(),
  inspected_at: z.string(),
  photo_ids: z.union([z.array(z.string()), z.null()]),
  reference_attachment_ids: z.union([z.array(z.string()), z.null()]),
  voice_note_id: z.union([z.string().uuid(), z.null()]),
  created_at: z.string(),
  updated_at: z.string(),
});
export type ChecklistItemResult = z.infer<typeof ChecklistItemResultSchema>;

export const ChecklistItemResultListSchema = z.array(ChecklistItemResultSchema);

export const InspectionSummarySchema = z.object({
  total_items: z.number().int(),
  completed: z.number().int(),
  passed: z.number().int(),
  failed: z.number().int(),
  not_applicable: z.number().int(),
  remaining: z.number().int(),
});
export type InspectionSummary = z.infer<typeof InspectionSummarySchema>;

export const ResultCreateSchema = z.object({
  verdict: InspectionVerdictEnum,
  note: z.union([z.string().max(4000), z.null()]).optional(),
  photo_ids: z.union([z.array(z.string()), z.null()]).optional(),
  reference_attachment_ids: z.union([z.array(z.string()), z.null()]).optional(),
});
export type ResultCreateInput = z.infer<typeof ResultCreateSchema>;

File: apps/api/src/bimstitch_api/schemas/inspection.py
from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field

from bimstitch_api.models.checklist_item_result import InspectionVerdict


class ResultCreate(BaseModel):
    verdict: InspectionVerdict
    note: str | None = Field(default=None, max_length=4000)
    photo_ids: list[str] | None = None


class ChecklistItemResultRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    checklist_item_id: UUID
    borgingsmoment_id: UUID
    project_id: UUID
    verdict: InspectionVerdict
    note: str | None
    inspector_user_id: UUID
    inspected_at: datetime
    photo_ids: list[str] | None
    voice_note_id: UUID | None
    created_at: datetime
    updated_at: datetime


class InspectionSummaryRead(BaseModel):
    total_items: int
    completed: int
    passed: int
    failed: int
    not_applicable: int
    remaining: int
